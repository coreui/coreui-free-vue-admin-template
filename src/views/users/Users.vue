<template>
  <b-row>
    <b-col cols="12" xl="6">
      <transition name="slide">
      <b-card :header="caption">
        <b-table :hover="hover" :striped="striped" :bordered="bordered" :small="small" :fixed="fixed" responsive="sm" :items="items" :fields="fields" :current-page="currentPage" :per-page="perPage" @row-clicked="rowClicked">
          <template slot="id" slot-scope="data">
            <strong>{{data.item.id}}</strong>
          </template>
          <template slot="name" slot-scope="data">
            <strong>{{data.item.name}}</strong>
          </template>
          <template slot="status" slot-scope="data">
            <b-badge :variant="getBadge(data.item.status)">{{data.item.status}}</b-badge>
          </template>
        </b-table>
        <nav>
          <b-pagination size="sm" :total-rows="getRowCount(items)" :per-page="perPage" v-model="currentPage" prev-text="Prev" next-text="Next" hide-goto-end-buttons/>
        </nav>
      </b-card>
      </transition>
    </b-col>
  </b-row>
</template>

<script>
import usersData from './UsersData'
export default {
  name: 'Users',
  props: {
    caption: {
      type: String,
      default: 'Users'
    },
    hover: {
      type: Boolean,
      default: true
    },
    striped: {
      type: Boolean,
      default: true
    },
    bordered: {
      type: Boolean,
      default: false
    },
    small: {
      type: Boolean,
      default: false
    },
    fixed: {
      type: Boolean,
      default: false
    }
  },
  data: () => {
    return {
      items: usersData.filter((user) => user.id < 42),
      fields: [
        {key: 'id'},
        {key: 'name'},
        {key: 'registered'},
        {key: 'role'},
        {key: 'status'}
      ],
      currentPage: 1,
      perPage: 5,
      totalRows: 0
    }
  },
  computed: {
  },
  methods: {
    getBadge (status) {
      return status === 'Active' ? 'success'
        : status === 'Inactive' ? 'secondary'
          : status === 'Pending' ? 'warning'
            : status === 'Banned' ? 'danger' : 'primary'
    },
    getRowCount (items) {
      return items.length
    },
    userLink (id) {
      return `users/${id.toString()}`
    },
    rowClicked (item) {
      const userLink = this.userLink(item.id)
      this.$router.push({path: userLink})
    }

  }
}
</script>

<style scoped>
.card-body >>> table > tbody > tr > td {
  cursor: pointer;
}
</style>
