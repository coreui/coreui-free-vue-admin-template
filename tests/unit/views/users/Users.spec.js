import { createLocalVue, shallowMount } from '@vue/test-utils'
import CoreuiVue from '@coreui/vue'
import Users from '@/views/users/Users'
import VueRouter from 'vue-router';

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

localVue.use(CoreuiVue)

describe('Users.vue', () => {
  it('has a name', () => {
    expect(Users.name).toBe('Users')
  })
  it('has a created hook', () => {
    expect(typeof Users.data).toMatch('function')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Users,{
      localVue,
      router
    })
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Users', () => {
    const wrapper = shallowMount(Users,{
      localVue,
      router
    })
    expect(wrapper.is(Users)).toBe(true)
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Users, {
      localVue,
      router
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('should have methods', () => {
    const wrapper = shallowMount(Users,{
      localVue,
      router
    })

    expect(typeof Users.methods.userLink  ).toEqual('function')
    expect(Users.methods.userLink(42)).toBe('users/42')
    expect(typeof Users.methods.rowClicked ).toEqual('function')
    expect(wrapper.vm.rowClicked({id:42})).toBeUndefined()
    expect(typeof Users.methods.getBadge ).toEqual('function')
    expect(wrapper.vm.getBadge('Active')).toBe('success')
    expect(wrapper.vm.getBadge('Inactive')).toBe('secondary')
  })
})
