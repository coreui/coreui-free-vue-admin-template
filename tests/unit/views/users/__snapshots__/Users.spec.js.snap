// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users.vue renders correctly 1`] = `
<crow-stub
  gutters="true"
  tag="div"
>
  <ccol-stub
    col="12"
    tag="div"
    xl="8"
  >
    <ccard-stub
      name="slide"
    >
      <ccardheader-stub>
        
          Users
        
      </ccardheader-stub>
       
      <ccardbody-stub>
        <cdatatable-stub
          clickablerows="true"
          fields="[object Object],[object Object],[object Object],[object Object]"
          header="true"
          hover="true"
          index-column=""
          items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
          itemsperpage="5"
          pagination="[object Object]"
          responsive="true"
          sortervalue="[object Object]"
          striped="true"
        />
      </ccardbody-stub>
    </ccard-stub>
  </ccol-stub>
</crow-stub>
`;
